QA.py

```diff
@@ -161,6 +156,7 @@ def _get_property(prop, extracted_info):
     ent_string = extracted_info["normalized entity"]
     ent = tag_lookup[extracted_info['tag']]
     try:
+        db = NimbusMySQLAlchemy()
```

Comment:
@cameron-toy , you may want to review whether this is OK.

Reply:
This one I'm less sure breaks something - [this function](https://github.com/calpoly-csai/api/blob/63117cc5aec05bd7b7caad5248ad5aa53bc373ab/QA.py#L164) uses the db outside of the object I think.

Reply:
This definitely breaks something. I also believe it initializes a database for each QA object. And there are hundreds of them, so that's bad.

It's a tricky situation because as @snekiam said, database access functions call db and they exist outside of the class. I could make each QA object take a db object in the constructor and each database access function take a db instance as a parameter. In that case, the best place to instantiate the db is probably in the constructor of the Nimbus object in nimbus.py.

Reply:
We can get around this with a cache of some kind or singleton